### ISSUE TITLE
Keras-2 support

### ISSUE THREAD

**Issue by the-moliver** [farizrahman4u/keras-contrib#40]
Keras-2 is nearing completion so we should start porting everything to work with the new Keras-2 APIs. I will create a branch for this purpose. Please state what you're working on in this thread so people don't duplicate effort.

**Comment by tboquet** [comment]
I can work on the examples.

**Comment by the-moliver** [comment]
I'm working on the cosine layers.

**Comment by kemaswill** [comment]
I can also work on applications as well as examples together with @tboquet since these two directories are very related.

**Comment by tboquet** [comment]
@kemaswill cool!

**Comment by abhaikollara** [comment]
I'd like to work on the backends

**Comment by abhaikollara** [comment]
@farizrahman4u New PRs to keras-2 branch or master branch ? I see a lot of keras-2 update commits on the master branch

**Comment by farizrahman4u** [comment]
Oh, no. You guys were already working on a Keras-2 branch? I was unaware of it. Thats a lot of redundant work.

**Comment by kemaswill** [comment]
Not yet actually, just wonder if we should first discuss what is needed to do and then we can start it.

**Comment by tboquet** [comment]
Not sure, bu it would be better to keep working on a branch so the transition will be clean?

**Comment by farizrahman4u** [comment]
Master is now compatible with Keras2. All PRs to Master.

**Comment by farizrahman4u** [comment]
Current issues.. we have accuracy issues with Batchrenorm and Deconv3d, causing tests to fail.

**Comment by abhaikollara** [comment]
@farizrahman4u Do I have to redo the PRs ?

**Comment by farizrahman4u** [comment]
What we should we currently working on is getting the tests to pass. We are having numerical issues with Batchrenorm and deconv3d. I have converted 90% to Keras2. There might be somethings I missed (such as in examples). You can work on converting them to Keras2.

**Comment by kemaswill** [comment]
Sure, I will work on the examples today.

**Comment by abhaikollara** [comment]
Okay, but there are issues with tensorflow backend, you can't even import keras_contrib. I had already fixed it on the keras-2. Shall I redo it on master ?

**Comment by farizrahman4u** [comment]
I just pushed a commit. Can you try again? If it still doesn't work, PR to Master.
